pymc3/tests/test_distributions_random.py

```diff
@@ -1306,6 +1305,65 @@ class TestOrderedProbit(BaseTestDistribution):
     ]
 
 
+class TestWishart(BaseTestDistribution):
+    def wishart_rng_fn(self, size, nu, V, rng):
+        return st.wishart.rvs(np.int(nu), V, size=size, random_state=rng)
+
+    pymc_dist = pm.Wishart
+
+    V = np.eye(3)
+    pymc_dist_params = {"nu": 4, "V": V}
+    reference_dist_params = {"nu": 4, "V": V}
+    expected_rv_op_params = {"nu": 4, "V": V}
+    sizes_to_check = [None, (), 1, (1,), 5, (4, 5), (2, 4, 2)]
+    sizes_expected = [
+        (3, 3),
+        (3, 3),
+        (1, 3, 3),
+        (1, 3, 3),
+        (5, 3, 3),
+        (4, 5, 3, 3),
+        (2, 4, 2, 3, 3),
+    ]
+    reference_dist = lambda self: functools.partial(
+        self.wishart_rng_fn, rng=self.get_random_state()
+    )
+    tests_to_run = [
+        "check_rv_size",
+        "check_pymc_params_match_rv_op",
+        "check_pymc_draws_match_reference",
+    ]
+
+
+class TestMatrixNormal(BaseTestDistribution):
+
+    pymc_dist = pm.MatrixNormal
+
+    mu = np.random.random((3, 3))
+    row_cov = np.eye(3)
+    col_cov = np.eye(3)
+
+    pymc_dist_params = {"mu": mu, "rowcov": row_cov, "colcov": col_cov}
+    expected_rv_op_params = {"mu": mu, "rowcov": row_cov, "colcov": col_cov}
+
+    sizes_to_check = [None, (1), (2, 3)]
+    sizes_expected = [(3,), (1, 3), (2, 3, 3)]
+
+    tests_to_run = ["check_rv_size", "check_pymc_params_match_rv_op", "test_matrix_normal"]
+
+    def test_matrix_normal(self):
```

Comment:
Shoukld we remove this test / replace this test with `rng_fn` from MatrixNormal ?

Reply:
I think the current one is better, as it provides a more independent check that our `rng_fn` is doing the correct thing. The standard test is used when we are just relying on a standard implementation in `rng_fn` (i.e., just calling scipy / numpy, without many steps in it). Is the test passing?

Reply:
> Is the test passing?

It's flaky. Currently fails but passed before.

Reply:
You might need to check dimension by dimension. The default test flattens everything and that might not be so comparable. This issue also cropped in here: https://github.com/pymc-devs/pymc3/pull/4596#issuecomment-822589620

(I might also be completely off)

Reply:
Looks like that could work, I added it for now.